pool/session.go

```diff
@@ -72,10 +72,13 @@ func (p *Pool) withinContainerSession(
 		return nil
 	}
 
-	cacheKey := formCacheKey(fmt.Sprintf("%p", c), signer)
+	cacheKey := cacheKeyForSession(fmt.Sprintf("%p", c), signer, verb, containerID)
 
 	tok, ok := p.cache.Get(cacheKey)
```

Comment:
you have changed the key for `Get` operations but what about `Put`? how do values fill the cache?

Reply:
Not exactly. I changed the _key_ for the new interface method, which was synced with client.Client. All these methods use `withinContainerSession`. Old pool methods were left as is (with old `formCacheKey`) because we will remove them in the next iteration.

`formCacheKey` and `cacheKeyForSession` generate different keys, thus we are safe from keys/data conflicts.

Reply:
yes, that is ok, i just did not see that you fill the cache below:

```go
// cache the opened session
p.cache.Put(cacheKey, tok)
```

that is what my question was about